src/rules/select-style.ts

```diff
@@ -0,0 +1,62 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils'
+
+import { docsUrl, pipeableSelect, storeSelect } from '../utils'
+
+export const ruleName = 'select-style'
+
+export const methodSelectMessageId = 'storeSelect'
+export const operatorSelectMessageId = 'operatorSelect'
+
+export type MessageIds =
+  | typeof methodSelectMessageId
+  | typeof operatorSelectMessageId
+
+export const OPERATOR = 'operator'
+export const METHOD = 'method'
+
+type Options = [string]
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: ruleName,
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Possible Errors',
+      description: `Selectors can be used either with 'select' as a pipeable operator or as a method`,
+      recommended: false,
+    },
+    schema: [
+      {
+        enum: [OPERATOR, METHOD],
+      },
+    ],
+    messages: {
+      [methodSelectMessageId]:
+        'Selectors should be used with select method: this.store.select(selector)',
+      [operatorSelectMessageId]:
+        'Selectors should be used with the pipeable operator: this.store.pipe(select(selector))',
+    },
+  },
+  defaultOptions: [OPERATOR],
+  create: (context) => {
+    const mode: string = context.options[0] || OPERATOR
+    return {
+      [pipeableSelect](node: TSESTree.CallExpression) {
+        if (mode === METHOD) {
+          context.report({
+            node,
+            messageId: methodSelectMessageId,
+          })
+        }
+      },
+      [storeSelect](node: TSESTree.Identifier) {
+        if (mode === OPERATOR) {
+          context.report({
+            node,
+            messageId: operatorSelectMessageId,
+          })
+        }
+      },
+    }
+  },
```

Comment:
I wonder if the following would be faster and if it's worth it:
If I got the syntax right it will only add one selector.

```suggestion
 return {
      ...(mode === METHOD ? {[pipeableSelect](node: TSESTree.CallExpression) {
        context.report({
          node,
          messageId: methodSelectMessageId,
        })
      }} : {}),
      ...(mode === OPERATOR ? {[storeSelect](node: TSESTree.Identifier) {
        context.report({
          node,
          messageId: operatorSelectMessageId,
        })
      }} : {}),
}
```

Reply:
It would probably be a bit faster but it's much harder to read imho. 
I think it could also be more difficult to extend it by adding more options if required in the future.
All in all, I am not sure the performance gain is worth it.

We could also write it like this (which I find more readable):
```suggestion
create: (context, [{ mode }]) => {
    if (mode === METHOD) {
      return {
        [pipeableSelect](node: TSESTree.CallExpression) {
          context.report({
            node,
            messageId: methodSelectMessageId,
          })
        },
      }
    }

    if (mode === OPERATOR) {
      return {
        [storeSelect](node: TSESTree.Identifier) {
          context.report({
            node,
            messageId: operatorSelectMessageId,
          })
        },
      }
    }
  },
```
But I never saw any rule written like this in the few I have read so I guess it's not really in the ESLint spirit.

Reply:
I am really not sure the performance difference would be noticeable.
Do you know any way we can do performance tests to get some data?
(if it's worth investing time in this)